services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -83,30 +83,49 @@ def adapt_links(self, content: dict, user: str) -> dict:
         links = content["links"]
         for i, link in enumerate(links):
             link_parser = urlparse(link["href"])
-            new_path = add_user_prefix(link_parser.path, user, "")
+            new_path = add_user_prefix(link_parser.path, user, collection_id)
             links[i]["href"] = link_parser._replace(path=new_path).geturl()
-        for i in range(len(content["collections"])):
-            content["collections"][i] = self.adapt_collection_links(content["collections"][i], user)
+        for i in range(len(content[object_name])):
+            content[object_name][i] = self.adapt_object_links(content[object_name][i], user)
         return content
 
     async def dispatch(self, request, call_next):
         """Redirect the user catalog specific endpoint and adapt the response content."""
-        request.scope["path"], user = remove_user_prefix(request.url.path)
+        ids = get_ids(request.scope["path"])
+        user = ids["owner_id"]
+        request.scope["path"] = remove_user_prefix(request.url.path)
+
+        if request.method in ["POST", "PUT"] and user:
+            request_body = await request.json()
+            if request.scope["path"] == "/collections":
+                request_body["id"] = f"{user}_{request_body['id']}"
+            elif f"/collections/{ids['owner_id']}_{ids['collection_id']}/items" in request.scope["path"]:
+                request_body["collection"] = f"{user}_{request_body['collection']}"
+            request._body = json.dumps(request_body).encode("utf-8")  # pylint: disable=protected-access
```

Comment:
This is pretty similar to changes from jira 25, we have to check if we are not doing the same thing. Also, we need a way to update request body, other than modifying some private variables.